File: Num8.js
let mainFunction = () => {
  console.log("mainFunction ->")
  let varA = 20
  console.log("varA: " +varA)
  let function2 = () => {
    console.log("function2 ->")
    varB = 50
    varA = 600
    console.log("varA: " +varA + "\nvarB: " + varB)
    let function3 = () => {
      console.log("function3 ->")
      varB = 900
      varC = 10
      console.log("varA: " +varA + "\nvarB: " + varB + "\nvarC: " + varC)
    }
    function3();
  }
  function2();
}
mainFunction();

File: Num8.py
def mainFunction():
  print("mainFunction ->")
  varA = 20
  print("varA: " + str(varA))
  def function2():
    print("function2 ->")
    global varA
    varB = 50
    varA = 600
    print("varA: " +str(varA) + "\nvarB: " + str(varB))
    def function3():
      print("function3 ->")
      global varB
      varB = 900
      varC = 10
      print("varA: " +str(varA) + "\nvarB: " + str(varB) + "\nvarC: " + str(varC))   
    function3()
  function2()
mainFunction()
